packages/react-aria-components/src/DragAndDrop.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import type {DropIndicatorProps as AriaDropIndicatorProps, ItemDropTarget} from 'react-aria';
+import type {DragAndDropHooks} from './useDragAndDrop';
+import type {DraggableCollectionState, DroppableCollectionState, MultipleSelectionManager} from 'react-stately';
+import React, {createContext, ForwardedRef, forwardRef, JSX, ReactNode, useCallback, useContext} from 'react';
+import type {RenderProps} from './utils';
+
+export interface DragAndDropContextValue {
+  dragAndDropHooks?: DragAndDropHooks,
+  dragState?: DraggableCollectionState,
+  dropState?: DroppableCollectionState
+}
+
+export const DragAndDropContext = createContext<DragAndDropContextValue>({});
+export const DropIndicatorContext = createContext<DropIndicatorContextValue | null>(null);
+
```

Comment:
Broke these out of useDragAndDrop.tsx into a separate file to improve tree shaking. Components depend on these contexts all the time but we shouldn't pull in the actual dnd hooks unless they are used.